### PR TITLE
Make move_metrics_to_cpu work recursively

### LINKED ISSUE #5831: move_metrics_to_cpu works non-recursively
## 🐛 Bug

https://github.com/PyTorchLightning/pytorch-lightning/blob/0b7f5a88a0f4691ec228c4708295a10d403fd592/pytorch_lightning/utilities/memory.py#L20-L46

`to_cpu` value should be passed to recursive calls.
```python
if isinstance(v, dict): 
    out_dict.update({k: recursive_detach(v, to_cpu=to_cpu)}) 
```

### Expected behavior

Should recursively detach (yes) and move tensors to cpu (bugged QQ).

### Environment

You can get the script and run it with:

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by ddrevicky**
## What does this PR do?
- Passes the `to_cpu` flag down the call chain in `recursive_detach`

Fixes #5831 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: tests/utilities/test_memory.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from pytorch_lightning.utilities.memory import recursive_detach
+
+
+def test_recursive_detach():
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    x = {"foo": torch.tensor(0, device=device), "bar": {"baz": torch.tensor(1.0, device=device, requires_grad=True)}}
+    y = recursive_detach(x, to_cpu=True)
+
+    assert x["foo"].device == device
+    assert x["bar"]["baz"].device == device
```
Comment (carmocca) [inline tests/utilities/test_memory.py:25]:
```suggestion
    device = "cuda" if torch.cuda.is_available() else "cpu"
    x = {"foo": torch.tensor(0, device=device), "bar": {"baz": torch.tensor(1.0, device=device, requires_grad=True)}}
    y = recursive_detach(x, to_cpu=True)

    assert x["foo"].device.type == device
    assert x["bar"]["baz"].device.type == device
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@ddrevicky mind rebase/merge master regarding updated TPU configuration... 🐰

**Review by tchaton — APPROVED** [review]
LGTM ! Great catch !
